cobbler/utils.py

```diff
@@ -216,9 +217,57 @@ def is_systemd() -> bool:
 
     This method currently checks if the path ``/usr/lib/systemd/systemd`` exists.
     """
-    if os.path.exists("/usr/lib/systemd/systemd"):
+    return os.path.exists("/usr/lib/systemd/systemd")
+
+
+def is_supervisord() -> bool:
+    """
+    Return whether or not this system uses systemd.
+
+    This method currently checks if the path ``/usr/bin/supervisord`` exists.
+    """
+    with ServerProxy('http://localhost:9001/RPC2') as server:
+        try:
+            server.supervisor.getState()
+        except OSError:
+            return False
         return True
-    return False
+
+
+def is_service() -> bool:
+    """
+    Return whether or not this system uses systemd.
+
+    This method currently checks if the path ``/usr/sbin/service`` exists.
+    """
+    return os.path.exists("/usr/sbin/service")
+
+
+def service_restart(service_name: str):
+    """
+    Restarts a daemon service independent of the underlining process manager. Currently SupervisorD, SystemD and SysV
+    are supported. Checks which manager is present is done in the order just described.
+
+    :param service_name: The name of the service
+    :returns: If the system is SystemD or SysV based the return code of the restart command.
+    """
+    if is_supervisord():
+        with ServerProxy('http://localhost:9001/RPC2') as server:
+            server.supervisor.stopProcess(service_name)
+            server.supervisor.startProcess(service_name)
+        return
+    elif is_systemd():
+        restart_command = "systemctl restart " + service_name
+    elif is_service():
+        restart_command = "service " + service_name + " restart"
+    else:
+        logger.warning("the service %s was not restarted", service_name)
```

Comment:
I think it's better to clarify that we cannot restart the service because there is no known service manager.